app/src/lib/git/diff.ts

```diff
@@ -367,14 +367,15 @@ export async function getWorkingDirectoryDiff(
     successExitCodes.add(1)
     args.push('--no-index', '--', '/dev/null', file.path)
   } else if (file.status.kind === AppFileStatusKind.Renamed) {
-    // NB: Technically this is incorrect, the best kind of incorrect.
-    // In order to show exactly what will end up in the commit we should
-    // perform a diff between the new file and the old file as it appears
-    // in HEAD. By diffing against the index we won't show any changes
-    // already staged to the renamed file which differs from our other diffs.
-    // The closest I got to that was running hash-object and then using
-    // git diff <blob> <blob> but that seems a bit excessive.
-    args.push('--', file.path)
+    // If there are unstaged changed, we can get staged differences from the
+    // original file or unstaged changes for the file. This prefers the
+    // unstaged changed.
+    if (file.status.hasUnstagedModifications) {
+      args.push('--', 'HEAD:' + file.status.oldPath, file.path)
```

Comment:
Does this make sense given https://github.com/desktop/desktop/pull/17364/files?diff=split&w=1#diff-93a97971dea4186b7445f010cdada081f05aeefd49b7b8900109e403d7b3a8ddR225 ?

I mean, it looks like with your changes, the app will never display the diff for a renamed file when there are unstaged changes.

Reply:
I struggled with figuring out the exact right thing to use here. I could get it to work if the file was moved with no changes and if a file was moved with changes and they were all staged. What I couldn't figure out was how to show a diff of a moved file (staged) with unstaged modified. So that's why I added that message. I figured it is better to know that the file was moved and has changes (even if Desktop doesn't display them) than have Desktop say it was moved with no changes when it actually had changed.

Do you have any suggestions on how I could figure out the diff for a renamed file when there are unstaged changes? I didn't want to automatically stage the changes. That would work, but if you're in that situation, you probably have manually staged files outside of Desktop, so you probably don't want Desktop staging additional changes.